include/zephyr/sensing/sensing_datatypes.h

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright (c) 2022-2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_SENSING_DATATYPES_H_
+#define ZEPHYR_INCLUDE_SENSING_DATATYPES_H_
+
+#include <stdint.h>
+
+/**
+ * @brief Data Types
+ * @addtogroup sensing_datatypes
+ * @{
+ */
+
+/**
+ * @struct sensing_sensor_value_header
+ * @brief sensor value header
+ *
+ * Each sensor value data structure should have this header
+ *
+ * Here use 'base_timestamp' (uint64_t) and 'timestampe_delta' (uint32_t) to
+ * save memory usage in batching mode.
+ *
+ * The 'base_timestamp' is for readings[0], the 'timestamp_delta' is relation
+ * to the previous 'readings'. So,
+ * timestamp of readings[0] is
+ *	header.base_timestamp + readings[0].timestamp_delta.
+ * timestamp of readings[1] is
+ *	timestamp of readings[0] + readings[1].timestamp_delta.
+ *
+ * Since timestamp unit is micro seconds, the max 'timestamp_delta' (uint32_t)
+ * is 4295 seconds.
+ *
+ * If a sensor has batched data where two consecutive readings differ by
+ * more than 4295 seconds, the sensor subsystem core will split them
+ * across multiple instances of the readings structure, and send multiple
+ * events.
+ *
+ * This concept is borrowed from CHRE:
+ * https://cs.android.com/android/platform/superproject/+/master:\
+ * system/chre/chre_api/include/chre_api/chre/sensor_types.h
+ */
+struct sensing_sensor_value_header {
+	/** base timestamp of this data readings, unit is micro seconds */
+	uint64_t base_timestamp;
+	/** count of this data readings */
+	uint16_t reading_count;
+};
```

Comment:
What do you think about adding a field into the header that identifies the sensor data type?

Reply:
I was thinking an identifier for `sensing_sensor_value_3d_int32`, `sensing_sensor_value_3d_int16`.  This is because the sensors readings are passed to the callback as a `void *`.  This would allow the application to validate the data format.

However, this may not be needed.  The callback already has a handle to the sensor that generated the data.  The application should know the expected data format.

An ID for the data type is not needed as part of this PR.  It's something to consider as future work.